### ISSUE TITLE
Fetch only one transit provider

### ISSUE THREAD

**Issue by danvk** [valhalla/mjolnir#473]
I'm running my own Valhalla instance for a small geographic region and would like to add transit. It seems like `valhalla_build_transit` fetches _all_ transit providers from transit.land. It would be nice if there were a way to tell it to only fetch a single one, or perhaps only providers within a bounding box.

**Comment by kevinkreiser** [comment]
@danvk yep bounding box(s) would be a great way to do this. I suppose we could also let you specify a list of operators or agencies as well as a means of filtering.

**Comment by danvk** [comment]
Either would be great. It looks like the transit.land API supports [both](https://transit.land/documentation/datastore/api-endpoints.html#format).

**Comment by gknisely** [comment]
@danvk 

@kevinkreiser is correct.  For now you could just look for particular transit providers.  You would have to update the which_tiles function to grab the id from the /api/v1/feeds.geojson.  Then you could just grab a particular provider with a simple if statement.  Example://       "id": "f-9q9-bart"  --> BART transit in SF.

**Comment by danvk** [comment]
How long does it typically take to fetch all the feeds? I'm about 30 minutes in. I'd prefer not to hack the source if it'll wrap up soon :)

**Comment by danvk** [comment]
Ah, this might not be optional after all. I'm getting HTTP 429s and transit.land API keys are apparently only available to MapZen employees. I'll see what I can do.

**Comment by kevinkreiser** [comment]
you should be able to do it without a key however we'd need to put in some rate limiting client side to avoid issues. feel free to make another issue for that if you'd like to go that route

**Comment by danvk** [comment]
Here's my patch (I'm using a slightly older version of mjolnir):

```
$ git diff
diff --git a/src/mjolnir/transit_fetcher.cc b/src/mjolnir/transit_fetcher.cc
index 39355c6..04ed4a0 100644
--- a/src/mjolnir/transit_fetcher.cc
+++ b/src/mjolnir/transit_fetcher.cc
@@ -127,6 +127,15 @@ std::priority_queue<weighted_tile_t> which_tiles(const ptree& pt) {
   curler_t curler;
   auto feeds = curler(url("/api/v1/feeds.geojson?", pt), "features");
   for(const auto& feature : feeds.get_child("features")) {
+    // limit to just Sun Metro
+    std::string id = feature.second.get<std::string>("id", "");
+    if (id != "f-9tsk-sunmetro") {
+      LOG_WARN("Skipping " + id);
+      continue;
+    } else {
+      LOG_INFO("Fetching " + id);
+    }
+
     //should be a polygon
     auto type = feature.second.get_optional<std::string>("geometry.type");
     if(!type || *type != "Polygon") {
```

I think this will work... but I'm going to get 429s for the rest of the day w/o an API key. Will report back.
